### DISCUSSION TITLE
I want to use STUN for intranet penetration to stream, how can I change the ports other than 47989 on Sunshine?

### DISCUSSION THREAD

**Discussion by a13997020** [LizardByte/community#210] [Sunshine] [answered]
Because STUN intranet penetration assigns a random public port each time.

**Comment by ReenigneArcher** [comment, marked as answer]
https://docs.lizardbyte.dev/projects/sunshine/en/latest/about/advanced_usage.html#port

**Reply by a13997020** [reply]
Thank you for your answer. Can't I customize all the ports instead of just changing the HTTP port? Because the public port given by STUN NAT traversal is completely random. I am using the Windows version of Sunshine.

**Reply by ReenigneArcher** [reply]
No, all the ports are relative to the HTTP port.
